src/extension_commands.ml

```diff
@@ -153,109 +153,49 @@ module Holes_commands : sig
-           end; if the current position comes before or inside the first range,
-           then pick the last hole in file *)
-        List.last_exn rest
-      else
-        let rec find_prev_hole prev_range = function
-          | [] -> prev_range
-          | range :: rest -> (
-            if Range.contains range ~positionOrRange:(`Position current_pos)
-            then
-              prev_range
-            else
-              let start = Range.start range in
-              match Position.compare current_pos start with
-              | Ordering.Less -> prev_range
-              | Greater -> find_prev_hole range rest
-              | Equal ->
-                (* assert false because we check whether the current pos is in
-                   this range in the if-expr above *)
-                assert false)
-        in
-        find_prev_hole fst_range rest
-
-  (** We need to be able to specify the position from which we want to look for
-      the next typed hole. If the [position] field of the argument object is not
-      set, then we use the current position in the active text editor.
-
-      We reuse "jump to hole" commands in other contexts (e.g., jump to next
-      hole in the destructed code), so we add an argument where if there is no
-      hole to jump to, user doesn't get notified because they didn't request
-      this jump themselves.
-
-      Note that we put all the configuration values in an object that is passed
-      as the {i first} argument to the command because we want explicit names
-      for those configurations, so object keys are useful here.
-
-      {[
-        args := [] |
-              [ {
-                "position" : { "line": <int>; "character": <int> },
-                "notify-if-no-hole": <bool> (default = true)
-              } ]
-      ]} *)
-  type arguments =
-    { current_position : Position.t option
-          (** [position]: <Position> in the arg obj *)
-    ; notify_if_no_hole : bool
-          (** [notify-if-no-hole]: <bool> (default = true) *)
-    }
-
-  let parse_arguments args =
-    match args with
-    | [] -> { current_position = None; notify_if_no_hole = true }
-    | [ params_obj ] ->
-      let json = Jsonoo.t_of_js params_obj in
-      let current_position =
-        Jsonoo.Decode.(try_optional (field "position" Position.t_of_json) json)
-      in
-      let notify_if_no_hole =
-        Jsonoo.Decode.(try_default true (field "notify-if-no-hole" bool) json)
-      in
-      { current_position; notify_if_no_hole }
-    | _ ->
-      failwith
-        "Jump to Previous/Next Hole: incorrect params passed to the command"
+  let hole_not_found_msg = "No typed hole was found in this file"
+
+  (** Shows appropriate message for when the [ocaml-lsp] in use by the extension
+      doesn't support jumping to holes. *)
+  let ocaml_lsp_doesn't_support_holes instance ocaml_lsp =
+    let suggestion =
+      match
+        Ocaml_lsp.is_version_up_to_date ocaml_lsp
```

Comment:
Why are we going to version numbers to query for features? We're using capabilities elsewhere.

Reply:
We do use capabilities to track whether ocaml-lsp can handle the command (see code below). This piece of code is run when ocaml-lsp doesn't have that capability to determine whether we need to ask the user to upgrade their ocaml-lsp version or they are bounded by their ocaml version. 

```ocaml
| Some (_, ocaml_lsp)
        when not (Ocaml_lsp.can_handle_typed_holes ocaml_lsp) ->
        ocaml_lsp_doesn't_support_holes instance ocaml_lsp
```